### PR TITLE
Suppress on property name changed config

### LINKED ISSUE #539: Updating to the newest version is now causing SuppressPropertyChangedWarnings in an existing codebase
I upgraded a project from version `3.0.1` to `3.2.6` and suddenly I'm getting a lot of this new error.

```
warning : Fody/PropertyChanged: Type Test does not contain a [NAME] property with an injected change notification, and therefore the On[NAME]Changed method will not be called. You can suppress this warning with [SuppressPropertyChangedWarnings].
```

There's a lot of places where we use a `OnXXXChanged` name without it refering to a specific property.

It would be nice to be able to turn this off at an assembly level, or in a config, rather than having to apply the attribute to every class.

### PR DESCRIPTION
**Pull request by distantcam**
#### Description

This PR fixes the missing config schema for `SuppressWarnings`
This PR adds a new config `SuppressOnPropertyNameChanged`

Fixes #539

#### Todos

 * [ ] The [options](https://github.com/Fody/PropertyChanged/wiki/Options) page needs to be updated

### CODE REVIEW

File: PropertyChanged.Fody/WarningChecker.cs [hunk 1]
```diff
@@ -67,7 +67,7 @@ public void EmitWarning(string message)
 
     public void EmitConditionalWarning(ICustomAttributeProvider member, string message)
     {
-        if (SuppressWarnings)
+        if (SuppressWarnings || SuppressOnPropertyNameChangedWarning)
```
Comment (ltrzesniewski) [inline PropertyChanged.Fody/WarningChecker.cs]: That doesn't look like the right place to check for the new flag. I'd check it at the call sites instead, as this is a generic method to emit all kinds of warnings.

Comment (distantcam) [reply PropertyChanged.Fody/WarningChecker.cs]: You're right. This was a rushed change. Fixed now.

**Comment by ltrzesniewski** [comment]
LGTM but for some reason the CI run timed out after one hour :/
I ran the tests locally and they all passed. Can you trigger the CI build again @SimonCropp?

**Comment by SimonCropp** [comment]
build passed

**Comment by distantcam** [comment]
I can't edit the options wiki myself, if you're waiting for that.

**Comment by ltrzesniewski** [comment]
No worries, I'll update the wiki page once this PR is merged.

Since you didn't merge it yet @SimonCropp I'll ask just in case: is this change OK for you?

**Comment by SimonCropp** [comment]
yep. lgtm

**Comment by ltrzesniewski** [comment]
I noticed a little detail (unfortunately after merging), when editing the wiki page: shouldn't we rename `SuppressOnPropertyNameChangedWarning` to `SuppressOnPropertyNameChangedWarnings` for consistency with `SuppressWarnings`? That would need to be done before a release.

**Pull request merged**